src/bson_util/async_encoding.rs

```diff
@@ -1,27 +1,32 @@
-use bson::{DecoderResult, Document, EncoderResult};
-use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
-use tokio_byteorder::{AsyncReadBytesExt, AsyncWriteBytesExt, LittleEndian};
+use bson::Document;
+use futures::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+
+use crate::{
+    error::Result,
+    runtime::{AsyncLittleEndianRead, AsyncLittleEndianWrite},
```

Comment:
`futures::io::Async{Read|Write}Ext` don't have the helpers for each integer type that tokio does, so I implemented them in the driver. 

This also helps guarantee that we use little endian bye order for all our reading and writing, which I think we were not doing in `cmap/conn/wire/message.rs` due to tokios helpers for these all reading/writing in big endian. The places where we were explicitly specifying little endian were of course already okay. I think the reason it didn't show up in any of our testing is that we were only using big endian for the flag bits and the checksum, neither of which we actually use regularly (at all?).

Reply:
Do we still want top keep these even though we're continuing to use tokio? The benefit you pointed out still applies, so I'm not opposed to keeping them

Reply:
I think we could still use tokio_byteorder if we wanted, given that we implement the tokio traits. It's not huge deal to implement our own version though, so I'm fine with leaving it.